### ISSUE TITLE
Improve frontend formatting

### ISSUE THREAD

**Issue by CohenArthur** [Rust-GCC/gccrs#1639]
- [ ] Make sure of newline between two function definitions
- [ ] When creating a scope in a switch case, brackets go under the colon and are indented
```c++
case foo: {
    auto bar = baz();
}
```
becomes
```c++
case foo:
  {
    auto bar = baz();
  }
```
- [ ] Make sure to respect the GNU coding style convention overall

**Comment by ArshErgon** [comment]
do we have to update the syntax you have provided for every single file? and will the new syntax will stay the same if some do `cargo fmt`?

**Comment by bjorn3** [comment]
Cargo fmt won't have any effect as gccrs is written in C++ and rustfmt only formats rust code.

**Comment by CohenArthur** [comment]
> do we have to update the syntax you have provided for every single file? and will the new syntax will stay the same if some do `cargo fmt`? please assign me for this task

Yes, and as @bjorn3 mentionned this is a C++ project so we're using `clang-format`. The syntax should stay the same when running `clang-format` as it's still valid GNU C++ style. Are you still interested in working on the issue @ArshErgon ?

**Comment by ArshErgon** [comment]
Hey @CohenArthur it will take some time, but I will do it.
